libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -144,12 +149,12 @@ public void onSaveInstanceState(Bundle outState) {
    * @param savedInstanceState to extract state variables
    */
   public void onRestoreInstanceState(Bundle savedInstanceState) {
-    NavigationViewInstanceState navigationViewInstanceState = savedInstanceState
-      .getParcelable(getContext().getString(R.string.instance_state));
-
+    String instanceKey = getContext().getString(R.string.navigation_view_instance_state);
+    NavigationViewInstanceState navigationViewInstanceState = savedInstanceState.getParcelable(instanceKey);
     recenterBtn.setVisibility(navigationViewInstanceState.getRecenterButtonVisibility());
     resetBottomSheetState(navigationViewInstanceState.getBottomSheetBehaviorState());
     updateInstructionListState(navigationViewInstanceState.isInstructionViewVisible());
+    mapInstanceState = savedInstanceState.getParcelable(MAP_INSTANCE_STATE_KEY);
```

Comment:
Have we considered adding this into `NavigationMapboxMap` or passing the key to `NavigationMapboxMap#onSaveInstanceState` somehow?
Having `MAP_INSTANCE_STATE_KEY` as `public` in `NavigationMapboxMap` looks weird to me.